sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1Test.java

```diff
@@ -304,17 +388,30 @@ public void testDeleteEntityDisplayData() {
   public void testDeleteKeyDoesNotAllowNullProject() throws Exception {
     thrown.expect(NullPointerException.class);
     thrown.expectMessage("projectId");
+    DatastoreIO.v1().deleteKey().withProjectId((String) null);
+  }
 
-    DatastoreIO.v1().deleteKey().withProjectId(null);
+  @Test
+  public void testDeleteKeyDoesNotAllowNullProjectValueProvider() throws Exception {
```

Comment:
These tests for "we bail on null" and "we don't bail on non-null", too, repeat a lot more times than it looks like they should have. Do they address a significant testing need? Is it possible to refactor the code so that you only need one or two of these? Do they overlap with other tests? (e.g. "we don't bail on non-null" is, I think, implicitly tested by every other test which doesn't purposely set things to null, so there's no risk of getting it wrong without noticing - it doesn't need its own test)